runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -139,12 +139,18 @@ public RuleContext addChild(RuleContext ruleInvocation) {
 		return addAnyChild(ruleInvocation);
```

Comment:
You don't want to set the parent of the ruleInvocation context like you do for terminal and error nodes?